### ISSUE TITLE
"Auto choose: true" not working?

### ISSUE THREAD

**Issue by Swede37** [Staartvin/Autorank-2#598]
Hi,

We've been using Autorank  all along as our ranking plugin, and previously when a player performed the ./ar check command it would show the requirements needed to rankup on their path, regardless whether they were basic players or had donor / staff permgroups as well. This was great, **exactly** what we want!

However now, (I'm thinking it's since 4.1) a player with additional perm groups will receive a message about [multiple paths](https://i.imgur.com/YzMpf8p.png), instead. It's almost like the "Auto choose: true"
isn't being recognized?  Here is a [config example](https://hastebin.com/qurugosami.cs), can you see anything wrong with this?

Currently using:
- PluginLibrary v1.1
- Statz v1.4.5
- Autorank 4.1-Dev21 (trying Dev version as the 4.1 release had the same issues)
- Spigot-4bd94dc-fb414fe (MC: 1.12.2) (Implementing API version 1.12.2-R0.1-SNAPSHOT)
- PermissionsEX
 
Any help would be greatly appreciated!

**Comment by Staartvin** [comment]
What happens if you remove the priority, but keep the auto choose option?

**Comment by Swede37** [comment]
Nope, that didn't seem to make any difference :(

I had troubles with using MySQL too, had to go back to flatfile.  _I'm cursed._

Any other thoughts?

**Comment by Swede37** [comment]
Is it possible it has something to do with the Essentials updates, or newer Spigot?  I'm using the pre-4.1 version on a server with older jars, and its working perfectly

**Comment by Staartvin** [comment]
Wait, so when you give a player a different group, it still works. It is just a few staff players that it does not work for?

**Comment by Swede37** [comment]
Unfortunately, it appears to be some others as well.  This actually only came to my attention a short while ago, as one of our Beta-testers mentioned it not working, and then all the staff chimed in that it never had worked and they just assumed it was because of the staff grouping.  It appears very low level ranks work, and somewhere along the way the ./ar check feature brings up the multiple grouping. I'm not sure what triggers it.  Would you mind if I sent you a copy of the complete config?

**Comment by Staartvin** [comment]
Hmm, it seems better if we can test this at your local server. I can't replicate your exact setup. Do you have any test server with the exact setup where we can test Autorank in isolation?

**Comment by Swede37** [comment]
You're more than welcome to stop by my server, it's in development mode so it's whitelisted.  Is your IGN Staartvin as well?

**Comment by Staartvin** [comment]
Yes , that's correct. What's the IP? I'm quite busy with work, but I can see if I can make time tomorrow. What timezone are you in?

**Comment by Staartvin** [comment]
Might be good to use some messaging service like Skype or Discord for instant messaging as well

**Comment by Swede37** [comment]
I'm in CDT- it's 15:27 here right now.  
I agree, Skype ok with you?  I just sent you a message

**Comment by Staartvin** [comment]
It's CET here (22:27). Hopefully I can get back to you tomorrow ;)

**Comment by Swede37** [comment]
Thanks, it's appreciated!  :)

**Comment by Swede37** [comment]
**We resolved this, thanks to Staartvin!**

So what we noticed was that _/ar check_  showed multiple paths, and that we had to choose one. Having the player do _/ar view list_ showed all the paths setup, and we were not setup for infinite pathing, so it should have only shown the one path available to that player at that time.  As explained to me, 

_"By default, Autorank will show all paths that exist for a player. If we turn on that option 'show based on prerequisites', Autorank will only show that path if the player meets the prerequisites of that path"_.  

**I added the the 'show_based_on_prerequisites: true' under the options, as suggested, restarted the server, and it was working!**

If this does not fix it for certain players, they will need to reset their path to clear out the previous information that shouldn't be there.  
- To do this, _/ar reset [player] [action]_ will reset the data of a given player (either their progress, chosen path or completed paths). 
- If the player then performs the _/ar check_ command, it will rank them up to where they should be. (If they had several ranks to go, they may need to perform the _/ar check_ a few times).
